internal/cmd/signer.go

```diff
@@ -13,9 +14,34 @@ import (
 
 // GetSigner returns a signer according to the CLI context.
 func GetSigner(opts *SignerFlagOpts) (notation.Signer, error) {
-	// Construct a signer from preconfigured key pair in config.json
-	// if key name is provided as the CLI argument
-	key, err := configutil.ResolveKey(opts.Key)
+	var key config.KeySuite
+	var err error
+
+	// Check if the options are valid for the key (Key is mutually exclusive with [KeyID, PluginName])
+	if opts.KeyID != "" && opts.PluginName != "" {
+		if opts.Key == "" {
+			// Construct a signer from on-demand key
+			mgr := plugin.NewCLIManager(dir.PluginFS())
+			plugin, err := mgr.Get(context.Background(), opts.PluginName)
+			if err != nil {
+				return nil, err
+			}
+			return signer.NewFromPlugin(plugin, opts.KeyID, map[string]string{})
+		} else {
+			return nil, errors.New("incompatible options, do not provide a key name when providing a key ID and plugin name")
+		}
+	} else if opts.KeyID == "" && opts.PluginName == "" {
+		// Construct a signer from preconfigured key pair in config.json
+		// if key name is provided as the CLI argument
+		key, err = configutil.ResolveKey(opts.Key)
+	} else {
+		if opts.Key == "" {
+			return nil, errors.New("incompatible options, both a key ID and plugin name are required when not using an existing key")
+		} else {
+			return nil, errors.New("incompatible options, do not provide a key ID or plugin name when providing a key name")
+		}
+	}
```

Comment:
This is difficult to read; can we move mutually exclusive validation to [PreRun or PreRunE](https://pkg.go.dev/github.com/spf13/cobra#section-readme) ?